src/Type/Definition/InputObjectType.php

```diff
@@ -91,10 +93,6 @@ public function getFields(): array
 
     protected function initializeFields(): void
     {
-        if (isset($this->fields)) {
```

Comment:
I intended to reduce complexity by putting the check here so it did not have to be repeated in other methods. Thought the method call was cheap and on some php optimization level this would have been inlined anyway I believe.

Reply:
In terms of opcodes, a method call is vastly more expensive than the single opcode of `isset()`.

The amount of code is reduced by putting the check inside the method, but I don't necessarily think the semantics are better. I generally prefer *does the thing need doing? -> do the thing* to *do the thing -> thing does nothing if it does not need doing*. It is somewhat comparable to having a method accept `null` as an argument and bail if it is passed - I would rather have the method be pure and check for `null` before.

Reply:
Yup, this is where we differ and we keep stumbling upon this over and over (😄) as I prefer to encapsulate checks within the method so I don't have to worry about them every time I call it. Though I get the approach and accent on performance in terms of this library.